akka-http-core/src/main/scala/akka/http/scaladsl/Transport.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.annotation.ApiMayChange
+import akka.http.scaladsl.Http.OutgoingConnection
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.scaladsl.{ Flow, Tcp }
+import akka.util.ByteString
+
+import scala.concurrent.Future
+
+/**
+ * Abstraction to allow the creation of alternative transports to run HTTP on.
+ */
+@ApiMayChange
+trait Transport {
```

Comment:
On the other hand, then we cannot expose it in `ConnectionPoolSettings` which is the main purpose of the whole thing.